ToolBox/BovespaDownloader/BovespaDataDownloader.cs

```diff
@@ -0,0 +1,449 @@
+                    var instrument = new LeanInstrument
+                    {
+                        Symbol = tokens[0],
+                        Name = tokens[1],
+                        Type = (SecurityType)Enum.Parse(typeof(SecurityType), tokens[2]),
+
+                    };
+
+                    if (tokens.Length >= 4) instrument.PointValue = double.Parse(tokens[3]);
+                    _instruments.Add(tokens[0], instrument);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Checks if downloader can get the data for the symbol
+        /// </summary>
+        /// <param name="symbol"></param>
+        /// <returns>Returns true if the symbol is available</returns>
+        public bool HasSymbol(Symbol symbol)
+        {
+            return _instruments.ContainsKey(symbol);
+        }
+
+        /// <summary>
+        /// Gets the security type for the specified symbol
+        /// </summary>
+        /// <param name="symbol">The symbol</param>
+        /// <returns>The security type</returns>
+        public SecurityType GetSecurityType(Symbol symbol)
+        {
+            return _instruments[symbol].Type;
+        }
+
+        /// <summary>
+        /// Get historical data enumerable for a single symbol, type and resolution given this start and end time (in UTC).
+        /// </summary>
+        /// <param name="symbol">Symbol for the data we're looking for.</param>
+        /// <param name="type">Security type</param>
+        /// <param name="resolution">Resolution of the data request</param>
+        /// <param name="startUtc">Start time of the data in UTC</param>
+        /// <param name="endUtc">End time of the data in UTC</param>
+        /// <returns>Enumerable of base data for this symbol</returns>
+        public IEnumerable<BaseData> Get(Symbol symbol, SecurityType type, Resolution resolution, DateTime startUtc, DateTime endUtc)
+        {
+            if (!_instruments.ContainsKey(symbol))
+                throw new ArgumentException("Invalid symbol requested: " + symbol);
+
+            if (type != SecurityType.Equity && type != SecurityType.Option && type != SecurityType.Future)
+                throw new NotSupportedException("SecurityType not available: " + type);
+
+            if (endUtc < startUtc)
+                throw new ArgumentException("The end date must be greater or equal to the start date.");
+
+            _inputDirectory = Directory.CreateDirectory(String.Format("{0}/{1}", Config.Get("input-directory", "./Data"), type)).FullName;
+            _ftpsite += (new Dictionary<SecurityType, string>() { { SecurityType.Future, "BMF" }, { SecurityType.Equity, "Bovespa-Vista" }, { SecurityType.Option, "Bovespa-Opcoes" } }[type]);
+
+            #region For equity daily files, use equity daily data!
+            if (type == SecurityType.Equity && resolution == Resolution.Daily)
+            {
+                for (var year = startUtc.Year; year <= endUtc.Year; year++)
```

Comment:
Yes! If one doesn't have the daily data (we need to download it manually at the Bovespa site, because of a captcha), daily Lean data can be built from the tick data.